patches/master_patch.patch

```diff
@@ -1831,6 +1879,207 @@ index a433afd0217830b49ddb3381ad55afeae2381840..ddcca749666f956c6317676dbcdd48da
+         break;
++      case STATE_AUTH:
++        rv = DoAuth(rv);
++        break;
+       case STATE_HANDSHAKE_WRITE:
+         DCHECK_EQ(OK, rv);
+         net_log_.BeginEvent(NetLogEventType::SOCKS5_HANDSHAKE_WRITE);
+@@ -274,8 +277,6 @@ int SOCKS5ClientSocket::DoLoop(int last_io_result) {
+   return rv;
+ }
+ 
+-const char kSOCKS5GreetWriteData[] = { 0x05, 0x01, 0x00 };  // no authentication
+-
+ int SOCKS5ClientSocket::DoGreetWrite() {
+   // Since we only have 1 byte to send the hostname length in, if the
+   // URL has a hostname longer than 255 characters we can't send it.
+@@ -285,8 +286,12 @@ int SOCKS5ClientSocket::DoGreetWrite() {
+   }
+ 
+   if (buffer_.empty()) {
+-    buffer_ = std::string(kSOCKS5GreetWriteData,
+-                          arraysize(kSOCKS5GreetWriteData));
++    const char greeting[] = {
++      0x05,			// SOCKS version
++      0x01,			// number of authentication methods
++      auth_method(),
++    };
++    buffer_ = std::string(greeting, sizeof(greeting));
+     bytes_sent_ = 0;
+   }
+ 
+@@ -345,14 +350,32 @@ int SOCKS5ClientSocket::DoGreetReadComplete(int result) {
+                       NetLog::IntCallback("version", buffer_[0]));
+     return ERR_SOCKS_CONNECTION_FAILED;
+   }
+-  if (buffer_[1] != 0x00) {
++  if (buffer_[1] != auth_method()) {
+     net_log_.AddEvent(NetLogEventType::SOCKS_UNEXPECTED_AUTH,
+                       NetLog::IntCallback("method", buffer_[1]));
+     return ERR_SOCKS_CONNECTION_FAILED;
+   }
+ 
+   buffer_.clear();
+-  next_state_ = STATE_HANDSHAKE_WRITE;
++  next_state_ = STATE_AUTH;
++  return OK;
++}
++
++int SOCKS5ClientSocket::DoAuth(int rv) {
++  rv = Authenticate(rv, *transport_, net_log_, io_callback_);
++  next_state_ = (rv == OK ? STATE_HANDSHAKE_WRITE : STATE_AUTH);
++  return rv;
++}
++
++uint8_t SOCKS5ClientSocket::auth_method() {
++  return 0x00;
++}
++
++int SOCKS5ClientSocket::Authenticate(int rv,
++                                     ClientSocketHandle& socket,
++                                     NetLogWithSource& net_log,
++                                     CompletionCallback& callback) {
++  DCHECK_EQ(OK, rv);
+   return OK;
+ }
+ 
+diff --git a/net/socket/socks5_client_socket.h b/net/socket/socks5_client_socket.h
+index afef312d001297ac1178b93fea92586102a54ac8..9c37710526d38ce9a48ef4942654e9dbaae2c567 100644
+--- a/net/socket/socks5_client_socket.h
++++ b/net/socket/socks5_client_socket.h
+@@ -19,6 +19,7 @@
+ #include "net/base/net_export.h"
+ #include "net/dns/host_resolver.h"
+ #include "net/log/net_log_with_source.h"
++#include "net/socket/socks_client_socket_pool.h"
+ #include "net/socket/stream_socket.h"
+ #include "url/gurl.h"
+ 
+@@ -40,7 +41,7 @@ class NET_EXPORT_PRIVATE SOCKS5ClientSocket : public StreamSocket {
+                      const HostResolver::RequestInfo& req_info);
+ 
+   // On destruction Disconnect() is called.
+-  ~SOCKS5ClientSocket() override;
++  virtual ~SOCKS5ClientSocket() override;
```

Comment:
the destructor is already virtual. I'm surprised the compiler doesn't complain about this

Reply:
So it is, can remove this patch.

Reply:
Fixed in https://github.com/brave/muon/pull/470/commits/98ea8e89fa14b7324fcab6f036857c9384d4b226.